tooling/minimize/src/main.rs

```diff
@@ -111,42 +117,88 @@ pub const DEFAULT_ARGS: &[&str] = &[
     "-Zub-checks=false",
 ];
 
-fn show_error(msg: &impl std::fmt::Display) -> ! {
+pub fn show_error(msg: &impl std::fmt::Display) -> ! {
     eprintln!("fatal error: {msg}");
     std::process::exit(101) // exit code needed to make ui_test happy
 }
 
+#[macro_export]
 macro_rules! show_error {
-    ($($tt:tt)*) => { crate::show_error(&format_args!($($tt)*)) };
+    ($($tt:tt)*) => {crate::show_error(&format_args!($($tt)*)) };
+}
+struct BeRustcCallbacks;
+
+impl rustc_driver::Callbacks for BeRustcCallbacks {}
```

Comment:
You can move these two inside the `be_rustc` function to make it clear that they are just used locally for that.